tests/test_adaptive_support.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import os
+from unittest import TestCase
+
+import numpy as np
+import pytest
+
+from aspire.denoising import adaptive_support
+from aspire.image import Image
+
+logger = logging.getLogger(__name__)
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "saved_test_data")
+
+
+class AdaptiveSupportTest(TestCase):
+    def setUp(self):
+        self.images = Image(
+            np.load(os.path.join(DATA_DIR, "sim_images_with_noise.npy"))
+        )
+
+    def test_adaptivate_support(self):
```

Comment:
Maybe add synthetic tests that test images of centered Gaussians? These have the advantage that we can calculate the correct `R` and `c` analytically (using `scipy.special.erf`).

Reply:
That's a great idea, thanks. I'll do this then hit the other items.

Reply:
I was able to make some progress here.  The real support seems to be working okay.   For a higher res image I am able to roughly estimate 1,2,3 standard deviations.  Example for centered gaussian disc with stddev 128 the code might return ~240 pixels for threshold 0.966 (two std devs).

I think may be an issue for the Fourier support values, or I'm deriving it wrong.  (By it, I mean the corresponding fourier transformed gaussian). TBD.

Reply:
I had the correct gaussian, talked about yesterday.  I've also tried two other functions from our meeting yesterday (`circ` and `hankel`).  I was able to incorporate the circ with the gaussian for one of the tests.

I was hoping since Hankel should be it's own transform that I could use this for a crude equivalence between Real and Fourier space, and test the F support, but it doesn't seem right to me.  Trying to figure that out now.

Reply:
Concretely, when I fftshift(fft2(hankel(...))), the resulting image is not as close to original as I thought it would be.

Reply:
I added two more functions for generating some test data to `misc`.  One for `circ` and another for `hankel`.  Now there are three synthetic tests:

1) Test that Fourier support of a Gaussian is similar to normal distribution.
2) Test that real support of a `hankel` image (1/(x**2 + y**2) and its Fourier transform are similar.
3) Test that Gaussian plus various `circ` radius achieve reasonable support values for 1,2,3,inf std deviations. Pretty loose, but looks that we are closer to the right std dev threshold.

Reply:
I added some comments in the new review. Resolving this one.